src/web-server/json-formatter.js

```diff
@@ -0,0 +1,34 @@
+import fs from '../lib/fs';
+
+export default class {
+    format404(url) {
+        return {
+            body: `The requested URL ${url} was not found on this server.`,
+            type: 'application/json',
+            status: 404,
+        };
+    }
+
+    formatDir(dirPath, entries) {
+        return {
+            body: JSON.stringify(entries),
+            type: 'application/json',
+            status: 200,
+        };
+    }
+
+    formatFile(path) {
+        return new Promise((resolve, reject) => {
+            fs.stat(path, (err, stats) => {
```

Comment:
Question: it seems like we have two layers of API going on here.  In `formatDir` we are passing the `entries`, which means `sh.ls` was invoked by the caller.  However, here, we're doing the `fs.stat` ourselves.  Can we pick one and do that in both cases?  Either move the `fs` handling in here or in the caller, but don't mix.